Sources/MapboxMaps/Gestures/GestureHandlers/DoubleTapGestureHandler.swift

```diff
@@ -0,0 +1,55 @@
+import UIKit
+
+/// `DoubleTapGestureHandler` updates the map camera in response
+/// to double tap gestures with 1 or 2 touches
+internal class DoubleTapGestureHandler: GestureHandler {
+
+    internal init(numberOfTouchesRequired: Int,
+                  gestureRecognizer: UITapGestureRecognizer,
+                  mapboxMap: MapboxMapProtocol,
+                  cameraAnimationsManager: CameraAnimationsManagerProtocol) {
+        precondition((1...2).contains(numberOfTouchesRequired))
+        gestureRecognizer.numberOfTapsRequired = 2
+        gestureRecognizer.numberOfTouchesRequired = numberOfTouchesRequired
+        super.init(
+            gestureRecognizer: gestureRecognizer,
+            mapboxMap: mapboxMap,
+            cameraAnimationsManager: cameraAnimationsManager)
+        gestureRecognizer.addTarget(self, action: #selector(handleGesture(_:)))
+    }
+
+    // Calls view to process the tap gesture
+    @objc private func handleGesture(_ gestureRecognizer: UITapGestureRecognizer) {
+        switch gestureRecognizer.state {
+        case .ended:
+            guard gestureRecognizer.numberOfTapsRequired == 2 else {
+                return
+            }
+            let gestureType: GestureType?
+            let zoomDelta: CGFloat?
+            switch gestureRecognizer.numberOfTouchesRequired {
+            case 1:
+                gestureType = .doubleTapToZoomIn
+                zoomDelta = 1
+            case 2:
+                gestureType = .doubleTapToZoomOut
+                zoomDelta = -1
+            default:
+                gestureType = nil
+                zoomDelta = nil
+            }
+            guard let gestureType = gestureType,
+                  let zoomDelta = zoomDelta else {
+                return
+            }
+            cameraAnimationsManager.cancelAnimations()
+            delegate?.gestureBegan(for: gestureType)
+            _ = cameraAnimationsManager.ease(to: CameraOptions(zoom: mapboxMap.cameraState.zoom + zoomDelta),
+                                             duration: 0.3,
+                                             curve: .easeOut,
+                                             completion: nil)
+        default:
+            break
```

Comment:
The idea of a switch inside of a switch looks a little gross. Maybe the code instead the `.ended` case can be moved to a separate private function, or internal if it is something that can be unit tested.

Also, switch is the convention when dealing with these states, but since you are only using one case, would this be a better case for an `if gestureRecognizer.state == .ended` ?

Reply:
It doesn't really bother me much, but I see your point. I'm happy to create a helper method here.

Reply:
Re: `if` vs `switch`, they both work. Let's keep it consistent across all the handlers.

Reply:
I'm good with keeping a `switch` I agree with the consistency. But yeah a helper method would be nice

Reply:
Updated to inject zoom delta and infer gesture type